main.go

```diff
@@ -26,36 +26,38 @@ func main() {
 	randomNumber := strconv.Itoa(int(GenerateRandomNumber()))
 
 	msgPtr := flag.String("m", randomNumber, "The message to send to the user")
-	phoneNumber := flag.String("n", "[+][country code][area code][local phone number]",
+	phoneNumber := flag.String("n", "+16366146678",
 		"The phone number you want to send message to in E.164 format")
 
 	flag.Parse()
 
 	if *msgPtr == "" || *phoneNumber == "" {
-		fmt.Println("You must supply a message and a phone number")
-		os.Exit(1)
+		log.Fatalf("You must supply a message and a phone number")
 	}
 
-	// Initialize a session that the SDK will use to load
-	// credentials from the shared credentials file. (~/.aws/credentials).
-	sess := session.Must(session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	}))
-
-	svc := sns.New(sess)
-
-	result, err := PublishMessage(svc, msgPtr, phoneNumber)
+	sess, err := NewAwsSession()
 	if err != nil {
-		fmt.Println("Got an error publishing the message:")
-		fmt.Println(err)
-		return
-	}
-	if err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+		log.Fatalln("Unable to create AWS session!", err)
 	}
 
-	fmt.Println(*result.MessageId)
+	log.Println("Vault client created.")
+	log.Println("AWS session created.")
+	log.Println("Running indefinitely...")
+
+	for {
```

Comment:
Do we have to run this indefinitely in our service?

Reply:
No, that is there just to test that credentials would be renewed appropriately over time.

You'll notice that the Vault client token renewal runs on context.Background().

Side note - this PR does not send a text message, just demonstrated valid token renewal.

Reply:
Sounds good then.